core/app/models/spree/promotion/rules/store.rb

```diff
@@ -0,0 +1,20 @@
+module Spree
+  class Promotion
+    module Rules
+      class Store < PromotionRule
+        has_many :promotion_rule_stores, class_name: "Spree::PromotionRuleStore",
+                                         foreign_key: :promotion_rule_id,
+                                         dependent: :destroy
+        has_many :stores, through: :promotion_rule_stores, class_name: "Spree::Store"
+
+        def applicable?(promotable)
+          promotable.is_a?(Spree::Order)
+        end
+
+        def eligible?(order, _options = {})
+          stores.none? || stores.include?(order.store)
```

Comment:
it seems really weird to check `stores.none?` here

Reply:
It is a little odd. I'm not sure why it was originally written this way in [`solidus_multi_domain`](https://github.com/solidusio/solidus_multi_domain/blob/master/app/models/spree/promotion/rules/store.rb#L11-L13), but I figured it would be best to keep it the same.